src/mds/SnapRealm.cc

```diff
@@ -454,15 +454,13 @@ snapid_t SnapRealm::resolve_snapname(std::string_view n, inodeno_t atino, snapid
   // first try me
   dout(10) << "resolve_snapname '" << n << "' in [" << first << "," << last << "]" << dendl;
 
-  //snapid_t num;
-  //if (n[0] == '~') num = atoll(n.c_str()+1);
-
   bool actual = (atino == inode->ino());
   string pname;
   inodeno_t pino;
   if (n.length() && n[0] == '_') {
-    int next_ = n.find('_', 1);
-    if (next_ > 1) {
+    size_t next_ = n.find_last_of('_');
+    if (next_ != std::string::npos &&
+	next_ > 1 && next_ + 1 < n.length()) {
```

Comment:
Why not just do `n.back() == '_'`?

Reply:
the format is _<snapname>_<inode number>